integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -282,20 +283,66 @@ fn fork_before_resharding_block(
     )
 }
 
-/// Returns a loop action that invokes a costly method from a contract multiple times per block height.
+enum ReceiptKind {
+    Delayed,
+    Buffered,
+}
+
+/// Checks that the shard containing `account` has a non empty set of receipts of type `kind` at the
+/// resharding block.
+fn check_receipts_presence_at_resharding_block(
+    account: AccountId,
+    kind: ReceiptKind,
+) -> LoopActionFn {
+    Box::new(
+        move |_: &[TestData],
+              test_loop_data: &mut TestLoopData,
+              client_handle: TestLoopDataHandle<ClientActorInner>| {
+            let client_actor = &mut test_loop_data.get_mut(&client_handle);
+            let tip = client_actor.client.chain.head().unwrap();
+
+            if !next_block_has_new_shard_layout(client_actor.client.epoch_manager.clone(), &tip) {
+                return;
+            }
+
+            let epoch_manager = &client_actor.client.epoch_manager;
+            let shard_id = epoch_manager.account_id_to_shard_id(&account, &tip.epoch_id).unwrap();
+            let shard_uid = &ShardUId::from_shard_id_and_layout(
+                shard_id,
+                &epoch_manager.get_shard_layout(&tip.epoch_id).unwrap(),
+            );
+            let congestion_info = &client_actor
```

Comment:
Along with checks for congestion_info I think what we should independently do is to verify whether the correct trie key entries are present or not.
- For delayed receipts we can check the `DelayedReceiptIndices` trie key and see if the start_index is not same as end_index.
- For buffered receipts it would be the `BufferedReceiptIndices` trie key and we can do something similar for the expected shard.

Reply:
Given this difference in checks, we can consider splitting checking delayed receipt and buffered receipt into separate functions if it looks more appropriate, else I'm fine with having them in the same function as well.

Reply:
Interesting! Let me see if I can expand this check easily (here in another PR)